xcode/Subconscious/Shared/Components/Common/TruncateWithGradientViewModifier.swift

```diff
@@ -18,54 +17,32 @@ struct TruncateWithGradientViewModifier: ViewModifier {
         content
             .frame(maxHeight: maxHeight, alignment: .topLeading)
             .clipped()
-            .overlay(
-                GeometryReader {
-                    geo in
+            .mask(
+                GeometryReader { geo in
                     if (geo.size.height >= maxHeight) {
-                        Rectangle()
-                            .fill(
-                                LinearGradient(
-                                    gradient: Gradient(
-                                        stops: [
-                                            Gradient.Stop(
-                                                color: color.opacity(0),
-                                                location: 0.1
-                                            ),
-                                            Gradient.Stop(
-                                                color: color,
-                                                location: 0.9
-                                            )
-                                        ]
-                                    ),
-                                    startPoint: .top,
-                                    endPoint: .bottom
-                                )
-                            )
-                            .frame(
-                                height: min(
-                                    maxHeight / 2,
-                                    Self.maxGradientHeight
-                                ),
-                                alignment: .bottom
-                            )
-                            .offset(
-                                y: geo.size.height - min(
-                                    maxHeight / 2,
-                                    Self.maxGradientHeight
-                                )
-                            )
-                            .allowsHitTesting(false)
+                        LinearGradient(
+                            gradient: Gradient(
+                                colors: [
+                                    .black,
+                                    .black,
+                                    .clear
+                                ]
+                            ),
+                            startPoint: .top,
+                            endPoint: .bottom
+                        )
+                    } else {
+                        Rectangle().fill(.black)
                     }
-                },
-                alignment: .bottom
+                }
```

Comment:
Reworked the gradient truncation to use a mask instead of gradient overlay. This is more robust / generalizable and _might_ even be more efficient to render.

Reply:
Awesome!